### PR TITLE
docs(tem): send email with api

### PR DESCRIPTION
**Pull request by nerda-codes**

### CODE REVIEW

**Review by bene2k1 — APPROVED** [review]

File: managed-services/transactional-email/api-cli/send-emails-with-api.mdx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  paragraph: This page explains how to send an email to multiple recipients using the Transactional Email API
+tags: transactional email send multiple-headers send-emails
+dates:
+  validation: 2024-04-24
+  posted: 2024-04-24
+categories:
+  - managed-services
+---
+
+This page shows you how to send a simple transactional email in `JSON` format to multiple recipients using the `additional_headers` parameter using the Scaleway Transactional Email API.
```
Comment (jcirinosclwy) [inline managed-services/transactional-email/api-cli/send-emails-with-api.mdx:16]:
```suggestion
This page shows you how to send a simple transactional email in `JSON` format to multiple recipients using the `additional_headers` parameter and the Scaleway Transactional Email API.
```

**Review by jcirinosclwy — APPROVED** [review]

File: managed-services/transactional-email/api-cli/send-emails-with-api.mdx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+1. Open a terminal and paste the following code to configure your environment variables. Make sure that you add your own values.
+
+        ```
+        export SCW_ACCESS_KEY="<API access key>"
+        export SCW_SECRET_KEY="<API secret key>"
+        export SCW_PROJECT_ID="<Scaleway Project ID>"
+        ```
+2. Run the following command to retrieve your domain's ID, as you will need it in the next step. The output should return your domain's information.
+        ```
```
Comment (jerjako) [inline managed-services/transactional-email/api-cli/send-emails-with-api.mdx:36]:
```suggestion
2. Run the following command to retrieve your domain's ID, as you will need it in the next step. The output should return your domain's information.

        ```
```

For consistency

File: managed-services/transactional-email/api-cli/send-emails-with-api.mdx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        curl -X GET \
+        -H "X-Auth-Token: $SCW_SECRET_KEY" \
+        "https://api.scaleway.com/transactional-email/v1alpha1/regions/fr-par/domains"
+        ```
+3. Run the following command to query your domain. Make sure that you replace `<domain-id>` with the ID of your domain.
+        ```
```
Comment (jerjako) [inline managed-services/transactional-email/api-cli/send-emails-with-api.mdx:42]:
```suggestion
3. Run the following command to query your domain. Make sure that you replace `<domain-id>` with the ID of your domain.

        ```
```

File: managed-services/transactional-email/api-cli/send-emails-with-api.mdx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+
+1. Open a terminal and paste the following code to configure your environment variables. Make sure that you add your own values.
+
+        ```
+        export SCW_ACCESS_KEY="<API access key>"
+        export SCW_SECRET_KEY="<API secret key>"
+        export SCW_PROJECT_ID="<Scaleway Project ID>"
+        ```
+2. Run the following command to retrieve your domain's ID, as you will need it in the next step. The output should return your domain's information.
```
Comment (jerjako) [inline managed-services/transactional-email/api-cli/send-emails-with-api.mdx:35]:
```suggestion
        ```

2. Run the following command to retrieve your domain's ID, as you will need it in the next step. The output should return your domain's information.
```

File: managed-services/transactional-email/api-cli/send-emails-with-api.mdx [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        ```
+        curl -X GET \
+        -H "X-Auth-Token: $SCW_SECRET_KEY" \
+        "https://api.scaleway.com/transactional-email/v1alpha1/regions/fr-par/domains"
+        ```
+3. Run the following command to query your domain. Make sure that you replace `<domain-id>` with the ID of your domain.
```
Comment (jerjako) [inline managed-services/transactional-email/api-cli/send-emails-with-api.mdx:41]:
```suggestion
        ```

3. Run the following command to query your domain. Make sure that you replace `<domain-id>` with the ID of your domain.
```

File: managed-services/transactional-email/api-cli/send-emails-with-api.mdx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+        ```
+        curl -X POST "https://api.scaleway.com/transactional-email/v1alpha1/regions/$REGION/domains/<domain-id>/check" \
```
Comment (jerjako) [inline managed-services/transactional-email/api-cli/send-emails-with-api.mdx:43]: This query is used to verify an unchecked domain. IDK if it's in the scope of this doc.

Either you need to add the step before (Create the domain via API)

Or remove this step and starts to actually sending an email with the API

File: managed-services/transactional-email/api-cli/send-emails-with-api.mdx [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+            "subject": "This is a subject", # Replace with your subject. Subjects must have at least 10 characters
+            "text": "This is a short sentence.", # Replace with the body of your email
+            "project_id": "<Scaleway Project ID>", # Replace '<Scaleway Project ID>' with your Scaleway Project ID
+            "attachments": [
+                {
+                    "name": "file.html", # Replace with the name of the file your want to send
+                    "type": "text/html; charset=\"utf8\"", # Replace with the attachment type you want to send
+                    "content": "PCFET0NUWVBFIGh0bWw+CjxodG1sIGxhbmc9ImVuIj4KICAgIDxoZWFkPgogICAgICAgIDxtZXRhIGNoYXJzZXQ9InV0Zi04Ij4KICAgICAgICA8dGl0bGU+dGl0bGU8L3RpdGxlPgogICAgPC9oZWFkPgogICAgPGJvZHk+CiAgICAgICAgPHA+SGVsbG8gPHNwYW4gc3R5bGU9ImNvbG9yOmJsdWUiPndvcmxkPC9zcGFuPiE8L3A+CiAgICA8L2JvZHk+CjwvaHRtbD4=" # Replace with the content you want to send
+                }
+            ],
```
Comment (jerjako) [inline managed-services/transactional-email/api-cli/send-emails-with-api.mdx:78]:
```suggestion
            "project_id": "<Scaleway Project ID>", # Replace '<Scaleway Project ID>' with your Scaleway Project ID
            "text": "This is a short sentence.", # Replace with the body of your email
            "html": "<html><body><p>This is a short sentence.</p></body></html>", # Replace with the content you want to send,
            "attachments": [
		{
			"name": "test.pdf",
			"type": "application/pdf",
			"content": "AAAA==" # Your PDF encoded in Base64
		}
            ],
```

This example of HTML file attachment, not the best one. Add an HTML in attachment is a really special case. You may want to add this HTML example in the "html" field, like this. Instead, a PDF kind be a good example.

File: managed-services/transactional-email/api-cli/send-emails-with-api.mdx [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+                }
+            ]
+        }
+        EOF
+
+        curl -X POST "https://api.scaleway.com/transactional-email/v1alpha1/regions/$REGION/emails" \
+            -H "X-Auth-Token: $SCW_SECRET_KEY" \
+            -d @mail.json
+        ```
+6. Once you have added your own information to the template above, run it in your terminal. An output similar to the following should display:
```
Comment (jerjako) [inline managed-services/transactional-email/api-cli/send-emails-with-api.mdx:92]:
```suggestion
        ```

6. Once you have added your own information to the template above, run it in your terminal. An output similar to the following should display:
```

File: managed-services/transactional-email/api-cli/send-emails-with-api.mdx [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+
+        ```
+        {"emails":[{"id":"655c27f2-b2a3-4a9f-8e1f-3e6dc268b1c4","message_id":"3d928e21-187a-4539-b303-403156e37911","project_id":"8j512135-9f5f-42b3-a900-9fdf0195b563","mail_from":"me@my.domain.example.com","rcpt_to":"your.recipient@mail.fr","rcpt_type":"to","created_at":"2024-04-01T07:55:36.758671147Z","updated_at":"2024-04-01T07:55:36.758671147Z","status":"new","status_details":"not yet processed","try_count":0,"last_tries":[]}]}%
```
Comment (jerjako) [inline managed-services/transactional-email/api-cli/send-emails-with-api.mdx:95]:
```suggestion
        {"emails":[{"id":"655c27f2-b2a3-4a9f-8e1f-3e6dc268b1c4","message_id":"3d928e21-187a-4539-b303-403156e37911","project_id":"8j512135-9f5f-42b3-a900-9fdf0195b563","mail_from":"me@my.domain.example.com","rcpt_to":"your.recipient@mail.fr","rcpt_type":"to","created_at":"2024-04-01T07:55:36.758671147Z","updated_at":"2024-04-01T07:55:36.758671147Z","status":"new","status_details":"not yet processed","try_count":0,"last_tries":[]}]}
```

File: managed-services/transactional-email/api-cli/send-emails-with-api.mdx [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+        ```
+
+7. Run the following command to check that your email has been sent. Make sure that you replace `email-id` with the ID of the email you retrieved in the output of the previous step.
+
+        ```
+        curl --request GET \
+        --url https://api.scaleway.com/transactional-email/v1alpha1/regions/fr-par/emails/email-id \
+        --header 'X-Auth-Token: $SCW_SECRET_KEY'
```
Comment (jerjako) [inline managed-services/transactional-email/api-cli/send-emails-with-api.mdx:103]:
```suggestion
7. Run the following command to check that your email has been sent. Make sure that you replace `$EMAIL_ID` with the ID of the email you retrieved in the output of the previous step.

        ```
        curl --request GET \
        --url https://api.scaleway.com/transactional-email/v1alpha1/regions/fr-par/emails/$EMAIL_ID \
        --header 'X-Auth-Token: $SCW_SECRET_KEY'
```

File: managed-services/transactional-email/api-cli/send-emails-with-api.mdx [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+        -H "X-Auth-Token: $SCW_SECRET_KEY" \
+        "https://api.scaleway.com/transactional-email/v1alpha1/regions/fr-par/domains"
+        ```
+
+3. Run the following command to query your domain. Make sure that you replace `<domain-id>` with the ID of your domain.
+
+        ```
+        curl -X POST "https://api.scaleway.com/transactional-email/v1alpha1/regions/$REGION/domains/<domain-id>/check" \
+            -H "X-Auth-Token: $SCW_SECRET_KEY" \
+            -d "{}"
```
Comment (nerda-codes) [inline managed-services/transactional-email/api-cli/send-emails-with-api.mdx:49]:
```suggestion
```

File: managed-services/transactional-email/api-cli/send-emails-with-api.mdx [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+            -d @mail.json
+        ```
+
+6. Once you have added your own information to the template above, run it in your terminal. An output similar to the following should display:
+
+        ```
+        {"emails":[{"id":"655c27f2-b2a3-4a9f-8e1f-3e6dc268b1c4","message_id":"3d928e21-187a-4539-b303-403156e37911","project_id":"8j512135-9f5f-42b3-a900-9fdf0195b563","mail_from":"me@my.domain.example.com","rcpt_to":"your.recipient@mail.fr","rcpt_type":"to","created_at":"2024-04-01T07:55:36.758671147Z","updated_at":"2024-04-01T07:55:36.758671147Z","status":"new","status_details":"not yet processed","try_count":0,"last_tries":[]}]}
+        ```
+
+7. Run the following command to check that your email has been sent. Make sure that you replace `$EMAIL_ID` with the ID of the email you retrieved in the output of the previous step.
```
Comment (nerda-codes) [inline managed-services/transactional-email/api-cli/send-emails-with-api.mdx:104]:
```suggestion
3. Run the following command to ensure that your domain is verified:

        ```
        curl -X GET "https://api.scaleway.com/transactional-email/v1alpha1/regions/$REGION/domains/<domain-id>" \
        -H "X-Auth-Token: $SCW_SECRET_KEY"
        ```

4. Copy the following template. Make sure that you replace the placeholder information with your own.

        ```
        cat > mail.json <<EOF
        {
            "from": {
                "name": "Me", # Replace 'Me' with your own name
                "email": "me@my.domain.example.com"  # Replace 'me@my.domain.example.com' with your email address
            },
            "to": [
                {
                    "name": "Your recipient", # Replace 'Your recipient' with your recipient's name
                    "email": "your.recipient@mail.fr" # Replace 'your.recipient@mail.fr' with your recipient's email address
                }
            ],
            "subject": "This is a subject", # Replace with your subject. Subjects must have at least 10 characters
            "project_id": "<Scaleway Project ID>", # Replace '<Scaleway Project ID>' with your Scaleway Project ID
            "text": "This is a short sentence.", # Replace with the body of your email
            "html": "<html><body><p>This is a short sentence.</p></body></html>", # Replace with the content you want to send,
            "attachments": [
		{
			"name": "test.pdf",
			"type": "application/pdf",
			"content": "AAAA==" # Your PDF encoded in Base64
		}
            ],
            "additional_headers": [
                {
                    "key": "Reply-To",
                    "value": "user1@example.com, user2@example.com, user3@example.com" # Replace the email addresses with the relevant ones
                }
            ]
        }
        EOF

        curl -X POST "https://api.scaleway.com/transactional-email/v1alpha1/regions/$REGION/emails" \
            -H "X-Auth-Token: $SCW_SECRET_KEY" \
            -d @mail.json
        ```

5. Once you have added your own information to the template above, run it in your terminal. An output similar to the following should display:

        ```
        {"emails":[{"id":"655c27f2-b2a3-4a9f-8e1f-3e6dc268b1c4","message_id":"3d928e21-187a-4539-b303-403156e37911","project_id":"8j512135-9f5f-42b3-a900-9fdf0195b563","mail_from":"me@my.domain.example.com","rcpt_to":"your.recipient@mail.fr","rcpt_type":"to","created_at":"2024-04-01T07:55:36.758671147Z","updated_at":"2024-04-01T07:55:36.758671147Z","status":"new","status_details":"not yet processed","try_count":0,"last_tries":[]}]}
        ```

6. Run the following command to check that your email has been sent. Make sure that you replace `$EMAIL_ID` with the ID of the email you retrieved in the output of the previous step.
```

**Pull request merged**